develop/blockchains/maintenance/storage-migrations.md

```diff
@@ -1,4 +1,183 @@
+
+- Read existing storage values in their original format
+- Transform data to match the new format
+- Write updated values back to storage
+- Calculate and return consumed weight
+
+### Migration Structure
+
+Storage migrations can be implemented using two approaches. The first method involves directly implementing `OnRuntimeUpgrade` on structs. This approach requires manually checking the on-chain storage version against the new [`StorageVersion`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/struct.StorageVersion.html){target=\_blank} and executing the transformation logic only when the check passes. This version verification prevents multiple executions of the migration during subsequent runtime upgrades.
+
+The recommended approach is to implement [`UncheckedOnRuntimeUpgrade`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.UncheckedOnRuntimeUpgrade.html){target=\_blank} and wrap it with [`VersionedMigration`](https://paritytech.github.io/polkadot-sdk/master/frame_support/migrations/struct.VersionedMigration.html){target=\_blank}. `VersionedMigration` implements `OnRuntimeUpgrade` and handles storage version management automatically, following best practices and reducing potential errors.
+
+`VersionedMigration` requires five type parameters:
+
+- `From` - the source version for the upgrade
+- `To` - the target version for the upgrade
+- `Inner` - the `UncheckedOnRuntimeUpgrade` implementation
+- `Pallet` - the pallet being upgraded
+- `Weight` - the runtime's [`RuntimeDbWeight`](https://paritytech.github.io/polkadot-sdk/master/frame_support/weights/struct.RuntimeDbWeight.html){target=\_blank} implementation
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/tags/polkadot-stable2409-1/substrate/frame/examples/single-block-migrations/src/migrations/v1.rs:0:122'
+```
+
+### Migration Organization
+
+Best practices recommend organizing migrations in a separate module within your pallet. Here's the recommended file structure:
+
+```plain
+my-pallet/
+├── src/
+│   ├── lib.rs       # Main pallet implementation
+│   └── migrations/  # All migration-related code
+│       ├── mod.rs   # Migrations module definition
+│       ├── v1.rs    # V0 -> V1 migration
+│       └── v2.rs    # V1 -> V2 migration
+└── Cargo.toml
+```
+
+This structure provides several benefits:
+
+- Separates migration logic from core pallet functionality
+- Makes migrations easier to test and maintain
+- Provides clear versioning of storage changes
+- Simplifies the addition of future migrations
+
+### Scheduling Migrations
+
+To execute migrations during a runtime upgrade, you need to configure them in your runtime's `Executive` pallet. Add your migrations in `runtime/src/lib.rs`:
+
+```rust
+--8<-- 'code/develop/blockchains/maintenance/storage-migrations/executive.rs'
```

Comment:
I'd also dive a bit deeper into the comment "More migration can be added here", since it's a really important point of the whole migration workflow